src/main/java/athena/TimeAllocator.java

```diff
@@ -23,35 +25,151 @@
      * @param tasks tasks in the current TaskList
      */
     public TimeAllocator(TaskList tasks) {
-        this.flexibleTask = tasks.getFilteredList(new FlexibleTimeFilter(true)).getTasks();
-        this.fixedTask = tasks.getFilteredList(new FlexibleTimeFilter(false)).getTasks();
+        this.flexibleTaskList = tasks.getFilteredList(new FlexibleTimeFilter(true));
+        this.fixedTaskList = tasks.getFilteredList(new FlexibleTimeFilter(false));
+        this.taskList = tasks;
+
+    }
+
+    public void runAllocate() throws TaskNotFoundException {
```

Comment:
Is this supposed to throw a `TaskNotFoundException`? If this throws a `TaskNotFoundException`, the user will receive an error message saying the task cannot be found. I think this would be really confusing to the user.